pkg/monitor/cloudinit.go

```diff
@@ -0,0 +1,97 @@
+package monitor
+
+import (
+	"context"
+	"path/filepath"
+	"time"
+
+	"github.com/fsnotify/fsnotify"
+	gocommon "github.com/harvester/go-common"
+	ctlnodev1 "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	"k8s.io/apimachinery/pkg/labels"
+
+	harvesterv1beta1 "github.com/harvester/node-manager/pkg/apis/node.harvesterhci.io/v1beta1"
+	"github.com/harvester/node-manager/pkg/cloudinit"
+	ctlharvesterv1beta1 "github.com/harvester/node-manager/pkg/generated/controllers/node.harvesterhci.io/v1beta1"
+)
+
+type CloudInitMonitor struct {
+	ctx             context.Context
+	monitorName     string
+	nodeName        string
+	cloudinits      ctlharvesterv1beta1.CloudInitController
+	cloudinitsCache ctlharvesterv1beta1.CloudInitCache
+	nodeCache       ctlnodev1.NodeCache
+}
+
+func NewCloudInitMonitor(ctx context.Context, monitorName, nodeName string, cloudinits ctlharvesterv1beta1.CloudInitController, nodeCache ctlnodev1.NodeCache) *CloudInitMonitor {
+	return &CloudInitMonitor{
+		ctx:             ctx,
+		monitorName:     monitorName,
+		nodeName:        nodeName,
+		cloudinits:      cloudinits,
+		cloudinitsCache: cloudinits.Cache(),
+		nodeCache:       nodeCache,
+	}
+}
+
+func (m *CloudInitMonitor) startMonitor() {
+	go func() {
+		gocommon.WatchFileChange(m.ctx, gocommon.FSNotifyHandlerFunc(m.handleFSNotify), []string{cloudinit.Directory})
+	}()
+}
+
+func (m *CloudInitMonitor) handleFSNotify(event fsnotify.Event) {
```

Comment:
Just have some questions about this event-driven case.
I saw we will return with some error path. Could we have a chance to retry?